homeassistant/components/lyric/climate.py

```diff
@@ -226,18 +226,29 @@ async def async_set_temperature(self, **kwargs) -> None:
         device = self.device
         if device.hasDualSetpointStatus:
```

Comment:
I think my problem lies here. My device shows `hasDualSetpointStatus: false`. Looking at the API docs, it sounds like maybe this related to `thermostatSetpointStatus` (TemporaryHold, HoldUntil, PermanentHold, VacationHold, NoHold) and not HeatSetpoint/CoolSetponit?

Here is what my device looks like in the Honeywell Developer Portal (with light redactions):
GET https://api.honeywell.com/v2/devices/thermostats/{device_id}
```json
{
  "groups": [
    {
      "id": 0,
      "name": "default",
      "rooms": [
        0,
        1,
        2
      ]
    }
  ],
  "displayedOutdoorHumidity": 91,
  "vacationHold": {
    "enabled": false
  },
  "currentSchedulePeriod": {
    "day": "Saturday",
    "period": "Home"
  },
  "scheduleCapabilities": {
    "availableScheduleTypes": [
      "None",
      "Geofenced",
      "TimedNorthAmerica"
    ],
    "schedulableFan": true
  },
  "scheduleType": {
    "scheduleType": "Timed",
    "scheduleSubType": "NA"
  },
  "inBuiltSensorState": {
    "roomId": 0,
    "roomName": "Kitchen"
  },
  "changeSource": {
    "by": "partner",
    "name": "Home Assistant"
  },
  "scheduleStatus": "Resume",
  "allowedTimeIncrements": 15,
  "settings": {
    "hardwareSettings": {
      "brightness": 0,
      "maxBrightness": 0
    },
    "fan": {
      "allowedModes": [
        "On",
        "Auto",
        "Circulate"
      ],
      "changeableValues": {
        "mode": "Auto"
      }
    },
    "temperatureMode": {
      "air": true
    },
    "specialMode": {
      "autoChangeoverActive": true,
      "emergencyHeatActive": false
    },
    "devicePairingEnabled": true
  },
  "deviceClass": "Thermostat",
  "deviceType": "Thermostat",
  "deviceID": "LCC-xxxxxxxx",
  "deviceInternalID": 000000,
  "userDefinedDeviceName": "Kitchen",
  "name": "Kitchen",
  "isAlive": true,
  "isUpgrading": false,
  "isProvisioned": true,
  "macID": "xxxxxxxxxxxxx",
  "deviceSettings": {},
  "priorityType": "FollowMe",
  "service": {
    "mode": "Up"
  },
  "deviceRegistrationDate": "2021-02-20T22:45:59.7033333",
  "dataSyncStatus": "Completed",
  "units": "Fahrenheit",
  "indoorTemperature": 68,
  "outdoorTemperature": 56.3,
  "allowedModes": [
    "EmergencyHeat",
    "Heat",
    "Off",
    "Cool",
    "Auto"
  ],
  "deadband": 0,
  "hasDualSetpointStatus": false,
  "minHeatSetpoint": 50,
  "maxHeatSetpoint": 90,
  "minCoolSetpoint": 50,
  "maxCoolSetpoint": 90,
  "changeableValues": {
    "mode": "Auto",
    "autoChangeoverActive": true,
    "emergencyHeatActive": false,
    "heatSetpoint": 68,
    "coolSetpoint": 78,
    "thermostatSetpointStatus": "NoHold",
    "nextPeriodTime": "21:30:00",
    "heatCoolMode": "Heat"
  },
  "operationStatus": {
    "mode": "EquipmentOff",
    "fanRequest": false,
    "circulationFanRequest": false
  },
  "indoorHumidity": 52,
  "indoorHumidityStatus": "Measured",
  "deviceModel": "T9-T10"
}

Reply:
I think you would have to use the current hvac mode to determine which setpoints should be adjusted. If hvac mode is Auto, then it should set heatSetpoint and coolSetpoint, otherwise, it should change the single setpoint temperature. Something like:
```python
if device.hvac_mode == HVAC_MODE_HEAT_COOL:
```

Not sure if hvac mode `off` would need any special handling? Right now it looks like it displays the single target temp on the dashboard, but should it display anything?

Hope that helps some!